python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1531,8 +1531,8 @@ def test_dataset_read_dictionary(tempdir, use_legacy_dataset):
     t1 = pa.table([[util.rands(10) for i in range(5)] * 10], names=['f0'])
     t2 = pa.table([[util.rands(10) for i in range(5)] * 10], names=['f0'])
     # TODO pass use_legacy_dataset (need to fix unique names)
-    pq.write_to_dataset(t1, root_path=str(path))
-    pq.write_to_dataset(t2, root_path=str(path))
+    pq.write_to_dataset(t1, root_path=str(path), use_legacy_dataset=True)
+    pq.write_to_dataset(t2, root_path=str(path), use_legacy_dataset=True)
```

Comment:
Same here as my comment above (https://github.com/apache/arrow/pull/12811/files#r845291213), as the TODO comment also indicates: calling this twice with `use_legacy_dataset=False` would overwrite the same file.

Reply:
Not sure why I didn't get this yesterday 🤦‍♀️

Reply:
Same here, this test can now be updated?